crates/cubecl-cuda/src/compiler/instruction.rs

```diff
@@ -463,3 +465,41 @@ impl Remainder {
         Ok(())
     }
 }
+
+macro_rules! unique_variable_name {
+    ($base:expr) => {
+        concat!("__", $base, "_", line!())
+    };
+}
+
+struct Normalize;
+
+impl Normalize {
+    fn format(
+        f: &mut core::fmt::Formatter<'_>,
+        input: &Variable,
+        out: &Variable,
+    ) -> core::fmt::Result {
+        let num = input.item().vectorization;
+        let norm = unique_variable_name!("norm");
```

Comment:
You can simply append "norm" to the output name:

```rust
let norm = format!("{out}_norm");
```

Putting things into another scope is also a good idea.